### PR TITLE
Exposebypasspropertychecks

### PR DESCRIPTION
**Pull request by bradtwurst**
We have the need to interrogate the current status of
'_bypassPropertyChecks' in classes that inherit from business base.

Attempting to interrogate the status using 'BypassPropertyChecks'
has the side effect of entering the bypass mode

### CODE REVIEW

**Comment by rockfordlhotka** [comment]
Is this not already exposed via an interface?

**Comment by bradtwurst** [comment]
I do not see anything that exposes the _bypassPropertyChecks value.  

There is a BypassPropertyChecks property, but it returns the BypassPropertyChecksObject.GetManager(this), which, upon construction, sets the _businessObject._bypassPropertyChecks value to true.

We are attempting to use the current value of _bypassPropertyChecks, but I don't see a way to observe it with affecting the value.

Please correct me if I am misreading the situation.

**Comment by rockfordlhotka** [comment]
Created #646 to track this issue.

File: Source/Csla.Shared/Core/BusinessBase.cs [hunk 1]
```diff
@@ -3598,6 +3598,17 @@ protected override void OnSetChildren(Csla.Serialization.Mobile.SerializationInf
     [NotUndoable]
     private bool _bypassPropertyChecks = false;
 
+    /// <summary>
+    /// Is the business object currently bypassing property checks?
```
Comment (rockfordlhotka) [inline Source/Csla.Shared/Core/BusinessBase.cs]: Should start with the phrase "Gets a value"

File: Source/Csla.Shared/Core/BusinessBase.cs [hunk 2]
```diff
@@ -3598,6 +3598,17 @@ protected override void OnSetChildren(Csla.Serialization.Mobile.SerializationInf
...
+    /// </summary>
+    protected internal bool IsBypassingPropertyChecks
+    {
+      get
+      {
+        return _bypassPropertyChecks;
```
Comment (rockfordlhotka) [inline Source/Csla.Shared/Core/BusinessBase.cs]: The get line returns a simple value - just put the whole thing on one line:

```
get { return _bypassPropertyChecks; }
```

File: Source/Csla.Shared/Core/BusinessBase.cs (hunk 1 above)
Comment (bradtwurst) [inline Source/Csla.Shared/Core/BusinessBase.cs]: updated

File: Source/Csla.Shared/Core/BusinessBase.cs [hunk 3]
```diff
@@ -3599,15 +3599,8 @@ protected override void OnSetChildren(Csla.Serialization.Mobile.SerializationInf
     private bool _bypassPropertyChecks = false;
 
     /// <summary>
-    /// Is the business object currently bypassing property checks?
     /// </summary>
```
Comment (rockfordlhotka) [inline Source/Csla.Shared/Core/BusinessBase.cs]: Did we totally lose the comment?

Comment (bradtwurst) [inline Source/Csla.Shared/Core/BusinessBase.cs]: Not in the code that I intended to commit.

However, the commit did not record the change properly.

An additional commit has been pushed for this line

**Pull request merged**